x-pack/plugins/security_solution/public/management/components/policy_response/policy_response_wrapper.test.tsx

```diff
@@ -0,0 +1,159 @@
+    malwareResponseConfigurations.status = overallStatus;
+    let downloadModelAction = policyResponse.Endpoint.policy.applied.actions.find(
+      (action) => action.name === 'download_model'
+    );
+
+    if (!downloadModelAction) {
+      downloadModelAction = {
+        name: 'download_model',
+        message: 'Failed to apply a portion of the configuration (kernel)',
+        status: overallStatus,
+      };
+      policyResponse.Endpoint.policy.applied.actions.push(downloadModelAction);
+    } else {
+      // Else, make sure the status of the generated action matches what was passed in
+      downloadModelAction.status = overallStatus;
+    }
+
+    if (
+      overallStatus === HostPolicyResponseActionStatus.failure ||
+      overallStatus === HostPolicyResponseActionStatus.warning
+    ) {
+      downloadModelAction.message = 'no action taken';
+    }
+
+    // Make sure that at least one configuration has the above action, else
+    // we get into an out-of-sync condition
+    if (malwareResponseConfigurations.concerned_actions.indexOf(downloadModelAction.name) === -1) {
+      malwareResponseConfigurations.concerned_actions.push(downloadModelAction.name);
+    }
+
+    // Add an unknown Action Name - to ensure we handle the format of it on the UI
+    const unknownAction: HostPolicyResponseAppliedAction = {
+      status: HostPolicyResponseActionStatus.success,
+      message: 'test message',
+      name: 'a_new_unknown_action',
+    };
+    policyResponse.Endpoint.policy.applied.actions.push(unknownAction);
+    malwareResponseConfigurations.concerned_actions.push(unknownAction.name);
+
+    return policyResponse;
+  };
+  const useGetEndpointPolicyResponseMock = useGetEndpointPolicyResponse as jest.Mock;
+  let render: () => ReturnType<AppContextTestRender['render']>;
+  const runMock = (customPolicyResponse?: HostPolicyResponse): void => {
+    useGetEndpointPolicyResponseMock.mockReturnValue({
+      data: { policy_response: customPolicyResponse ?? createPolicyResponse() },
+      isLoading: false,
+      isFetching: false,
+      isError: false,
+    });
+  };
+
+  beforeEach(() => {
+    const mockedContext = createAppRootMockRenderer();
+    render = () => mockedContext.render(<PolicyResponseWrapper endpointId="id" />);
+  });
+
+  it('should include the title', async () => {
+    runMock();
+    expect((await render().findByTestId('endpointDetailsPolicyResponseTitle')).textContent).toBe(
+      'Policy Response'
+    );
+  });
+
+  it('should display timestamp', () => {
+    runMock();
+    const timestamp = render().queryByTestId('endpointDetailsPolicyResponseTimestamp');
+    expect(timestamp).not.toBeNull();
+  });
+
+  it('should show a configuration section for each protection', async () => {
+    runMock();
+    const configAccordions = await render().findAllByTestId(
+      'endpointDetailsPolicyResponseConfigAccordion'
+    );
+    expect(configAccordions).not.toBeNull();
```

Comment:
🤔 is this valid? I thought that `findAll*()` methods will always return an array (which could be empty), so `not.toBeNull()` will always be true.
Maybe you need to check for `.hasLength(x)`?

This comment may apply to other usages you have below

Reply:
Good point, that's correct. This was ported from the endpoint details test, will fix those